Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp

```diff
@@ -687,7 +687,7 @@ void WebsiteDataStore::removeData(OptionSet<WebsiteDataType> dataTypes, WallTime
         }
     }
 
-    if (dataTypes.contains(WebsiteDataType::DeviceIdHashSalt) || (dataTypes.contains(WebsiteDataType::Cookies)))
+    if (dataTypes.contains(WebsiteDataType::DeviceIdHashSalt))
```

Comment:
I am not sure we should do this change.
Spec states: `[User Agents](https://w3c.github.io/mediacapture-main/#dfn-user-agent) MUST rotate per-origin device identifiers when other persistent storage are cleared.`

Reply:
ah so this suggests device id should have the same lifetime as cookies. In this case do we think HashSalt is cookies?
If it is:
Then we should return WKWebsiteDataTypeCookies for website data fetch when clients have DeviceIdHashSalt data, and remove both cookies and hash salt when clients remove WKWebsiteDataTypeCookies data.

If it is not:
Then we should return WKWebsiteDataTypeHashSalt for website data, and let client remove data for WKWebsiteDataTypeHashSalt. (We may keep the existing behavior that removes WKWebsiteDataTypeHashSalt data when clients remove WKWebsiteDataTypeCookies)

Which one do you think it should be?

Reply:
I would do the latter, expose it but continue removing it when cookies are cleared.

Reply:
Ok, the patch is updated as suggested